benchmarks/HOTPOTQA/hotpotqa_benchmark.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import os
+import time
+from typing import Dict, Tuple, Any
+
+from datasets import Dataset
+
+from benchmarks import BaseBenchmark
+from rageval.metrics import (AnswerEMCorrectness, AnswerF1Correctness)
+
+
+class HOTPOTQABenchmark(BaseBenchmark):
+    """Benchmark for HotPotQA dataset.
+
+    HotPotQA is a new dataset with 113k Wikipedia-based question-answer pairs with four key features: (1) the questions require finding and reasoning over multiple supporting documents to answer; (2) the questions are diverse and not constrained to any pre-existing knowledge bases or knowledge schemas; (3) we provide sentence-level supporting facts required for reasoning, allowingQA systems to reason with strong supervision and explain the predictions; (4) we offer a new type of factoid comparison questions to test QA systems’ ability to extract relevant facts and perform necessary comparison.
+
+    """
+
+    name = "hotpot_qa_benchmark"
+
+    def __init__(self) -> None:
+        """Initialization."""
+        super().__init__()
+
+    def _recode_gt_supporting_facts(self, data: object) -> object:
+        """To calculate f1 recode gt_sent_ids from 1 to the length of all sentences in contexts """
+        recode_answers = []
+        len_support = [len(i) for i in data['context']['sentences']]
+        for title, sent_id in zip(data['supporting_facts']['title'], data['supporting_facts']['sent_id']):
+            idx = data['context']['title'].index(title)
+            recode = sum(len_support[:idx]) + 1 + sent_id
+            recode_answers.append(str(recode))
+        recode_answers=[' '.join(recode_answers)]
+        data["gt_sent_ids"] = recode_answers
+        return data
+
+    def _evaluate(self) -> Tuple[Dict[Any, Any], Dataset]:
+        """Evaluate the dataset and return the dataset with scores.
+
+        For the HotPotQA dataset, the `short_answer` and `supporting_answer` which we need to extract
```

Comment:
For the HotPotQA dataset, we need to extract the `short_answer` and `supporting_answer`.